spec/integration/parser/environment_spec.rb

```diff
@@ -0,0 +1,46 @@
+require 'spec_helper'
+
+describe "A parser environment setting" do
+
+  let(:confdir) { Puppet[:confdir] }
```

Comment:
I think this needs to point to a tmp directory in order for the tests to not clutter the file system, then all the files can be regular files under that tmp directory (they will all then be cleaned up).